dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -32,8 +31,37 @@ namespace dali {
 namespace detail {
 namespace wds {
 
+/**
+ * The index version string should look like:
+ * v<one-or-two-digits>.<one-or-two-digits>
+ * e.g. v1.2, v10.1, v0.22, v12.12
+ */
+inline int DLL_PUBLIC CalcIndexVersion(const std::string& index_version_string) {
+  using std::isdigit;
+  auto it = index_version_string.rbegin();
+  int major = 0, minor = 0;
+  if (!isdigit(*it)) return -1;
+  minor += *it++ - '0';
+  if (isdigit(*it)) {
+    minor += (*it++ - '0') * 10;
```

Comment:
This limits us to just two digits _and_ is more convoluted than parsing left to right.

Reply:
Calculating version string like `major * 1000 + minor` limits minor version to two digits anyhow.

Reply:
It limits it to three digits.

Reply:
IIUC, it limits to two digits:
```
v11.10 -> 11 * 1000 + 10 = 1110
v1.110 -> 1 * 1000 + 110 = 1110
```
Assuming minor version has 3 digits, these two versions collide introducing a vulnerability. Therefore we need to assume 2-digit minor version. I've also assumed 2-digit major version, cause even if we change the version every week, the `v99.99` cap will be reached in 100 years or so.

Reply:
That's correct, my bad. But is the 2-digit limit bad? Even if we change the version every week, we'll get to `v99.99` in a 100 years or so.

Reply:
I'm not saying the limit itself is bad. It's rather that this kind of parsing is quite unusual. Were we going from start to end, this would be a simple matter of using `while` instead of `if`, like in the code I posted some time ago.